packages/actor-extract-links-extract-tree/lib/solver.ts

```diff
@@ -0,0 +1,395 @@
+      }
+      hasVariable = true;
+    } else if ('term' in arg && arg.term.termType === 'Literal') {
+      rawValue = arg.term.value;
+      valueType = SparqlOperandDataTypesReversed.get(arg.term.datatype.value);
+      if (valueType) {
+        valueAsNumber = castSparqlRdfTermIntoNumber(rawValue!, valueType);
+        if (!valueAsNumber) {
+          return new UnsupportedDataTypeError(`we do not support the datatype "${valueType}" in the solver for the moment`);
+        }
+      } else {
+        return new UnsupportedDataTypeError(`The datatype "${valueType}" is not supported by the SPARQL 1.1 Query Language W3C recommandation`);
+      }
+    }
+  }
+  // Return if a fully form solver expression can be created
+  if (hasVariable && rawValue && valueType && valueAsNumber) {
+    return {
+      variable,
+      rawValue,
+      valueType,
+      valueAsNumber,
+      operator,
+      chainOperator: linksOperator,
+    };
+  }
+  const missingTerm = [];
+  if (!hasVariable) {
+    missingTerm.push('Variable');
+  }
+  if (!rawValue) {
+    missingTerm.push('Litteral');
+  }
+
+  return new MisformatedFilterTermError(`the filter expressions doesn't have the term ${missingTerm.toString()}`);
+}
+
+/**
+ * Recursively traverse the filter expression and calculate the domain until it get to the current expression.
+ * It will thrown an error if the expression is badly formated or if it's impossible to get the solution range.
+ * @param {Algebra.Expression} filterExpression - The current filter expression that we are traversing
+ * @param {SolutionDomain} domain - The current resultant solution domain
+ * @param {LogicOperator} logicOperator - The current logic operator that we have to apply to the boolean expression
+ * @param {Variable} variable - The variable targeted inside the filter expression
+ * @param {boolean} notExpression
+ * @returns {SolutionDomain} The solution domain of the whole expression
+ */
+export function recursifResolve(
+  filterExpression: Algebra.Expression,
+  domain: SolutionDomain,
+  logicOperator: LogicOperator | undefined,
+  variable: Variable,
+  notExpression: boolean,
+): SolutionDomain {
+  // We apply an or operator by default or if the domain is empty
+  if (!logicOperator || domain.isDomainEmpty()) {
+    logicOperator = LogicOperator.Or;
+  }
+
+  if (filterExpression.expressionType === Algebra.expressionTypes.TERM
+  ) {
+    if (filterExpression.term.value === 'false') {
+      domain = domain.add({ range: A_FALSE_EXPRESSION, operator: logicOperator });
+    } else if (filterExpression.term.value === 'true') {
+      domain = domain.add({ range: A_TRUE_EXPRESSION, operator: logicOperator });
+    } else {
+      throw new MisformatedFilterTermError(`The term sent is not a boolean but is this value {${filterExpression.term.value}}`);
+    }
+  } else if (
+    // If it's an array of term then we should be able to create a solver expression
```

Comment:
I have no idea what is happing here. Could you improve the documentation or simplify the code?